qt/scientific_interfaces/Muon/ALCInterface.cpp

```diff
@@ -105,16 +104,16 @@ void ALCInterface::initLayout() {
   m_dataLoading->initialize();
   m_dataLoading->setParent(this);
 
-  m_baselineModellingView = new ALCBaselineModellingView(m_ui.baselineModellingView);
-  m_baselineModelling = new ALCBaselineModellingPresenter(m_baselineModellingView, m_baselineModellingModel);
+  auto baselineModellingModel = std::make_unique<ALCBaselineModellingModel>();
+  ALCBaselineModellingView *baselineModellingView = new ALCBaselineModellingView(m_ui.baselineModellingView);
+  m_baselineModelling = new ALCBaselineModellingPresenter(baselineModellingView, std::move(baselineModellingModel));
   m_baselineModelling->initialize();
 
   m_peakFittingView = new ALCPeakFittingView(m_ui.peakFittingView);
   m_peakFitting = new ALCPeakFittingPresenter(m_peakFittingView, m_peakFittingModel);
   m_peakFitting->initialize();
 
   connect(m_dataLoading, SIGNAL(dataChanged()), SLOT(updateBaselineData()));
-  connect(m_baselineModellingModel, SIGNAL(correctedDataChanged()), SLOT(updatePeakData()));
```

Comment:
I am a bit confused by how you can delete this line without somehow calling `updatePeakData()` inside of `setCorrectedData()` in the presenter. Before, when the model would call `setCorrectedData()`, the signal `correctedDataChanged()` would be picked up in two places, one inside the presenter that called `updateCorrectedCurve()` (which you have correctly accounted for), and another inside this ALCInterface that would call `updatePeakData()`, which I don't see how it's being done now. I haven't tested it manually yet, so maybe everything works anyway

Reply:
It looks like this is covered by line 278 to 280 for this file where it calls `updatePeakData` after doing `setCorrectedData`